### PR TITLE
downloads: don't remember dir for temp downloads

### LINKED ISSUE #2173: Don't remember temporary download directory
I have the following settings:

storage -> download-directory = ""
storage -> prompt-download-directory = true
storage -> remember-download-directory = true

I noticed something unusual about the suggested download directory:

1. run `:download` to open the download prompt
2. note the suggested download location is my home directory or something
   'normal'
3. type `<c-x>` to open the file instead of save
4. later run `:download` again on some page
5. note the suggested download location is `/tmp/qutebrowser-download-xxxxxx/`

I believe this to be a result of the `remember-download-directory` option remembering where the previously downloaded (opened) file was downloaded to (the temp dir) and suggesting that. 

The behaviour should be that it only remembers the download location of the previously _downloaded_ file (not downloaded to a temporary location to be opened).

### PR DESCRIPTION
**Pull request by Kingdread**
Fixes #2173.

See [this comment](https://github.com/The-Compiler/qutebrowser/issues/2173#issuecomment-268246936) for explanation.

### CODE REVIEW

**Comment by The-Compiler** [comment]
Can you look at adding a test for this in `downloads.feature`? I think it should be relatively straightforward.

File: qutebrowser/browser/downloads.py [hunk 1]
```diff
@@ -600,7 +603,8 @@ def _set_filename(self, filename, *, force_overwrite=False):
                                                   os.path.expanduser('~'))
 
         self.basename = os.path.basename(self._filename)
-        last_used_directory = os.path.dirname(self._filename)
+        if remember_directory:
+            last_used_directory = os.path.dirname(self._filename)
```
Comment (The-Compiler) [inline qutebrowser/browser/downloads.py]: Hmm, looks like the test still passes even when reverting this, which shouldn't happen? Not sure why, though I haven't taken a deeper look.

Comment (Kingdread) [inline qutebrowser/browser/downloads.py]: Strange, it fails locally when reverting the change... Are you by any chance running without `QUTE_BDD_WEBENGINE=true`? This issue only exists with the webengine backend, the test always passes with webkit.

**Comment by The-Compiler** [comment]
I indeed forgot `--qute-bdd-webengine` - sorry, my bad!

**Pull request merged**